modules/Bio/EnsEMBL/Production/Pipeline/Webdatafile/WebdataFileContigs.pm

```diff
@@ -119,9 +120,10 @@ sub map_data {
   my $old_version = $slice->coord_system()->version();
 
   my @decoded_segments;
-  my $projection = $slice->project('contig'); 
+  try {
```

Comment:
This is rather un-Perlish - an eval (https://users.cs.cf.ac.uk/Dave.Marshall/PERL/node114.html) would do much the same without an external library. I guess you've done this because errors arise if there are no 'contig' regions to project to - but if an error happens to occur for any other reason, execution will still continue, which seems undesirable. Maybe a check for a contig coord_system could be done instead?

Reply:
sure will do in perl way by using eval.  ya your right  exception code will not fail for any other error. i will add eval block for eval { $projection = $slice->project('contig') } ;  
  i will check contig coord_system to.